### PR TITLE
building for Raspberry Pi reference device

### PR DESCRIPTION
**Pull request by estenberg**
@drewmoseley @kacf @GregorioDiStefano please review.
I will backport to 1.3.x and 1.2.x once approved.

### CODE REVIEW

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md [hunk 1]
```diff
@@ -181,7 +203,7 @@ Once all the configuration steps are done, an image can be built with bitbake:
 bitbake <YOUR-TARGET>
 ```
 
-!!! Please replace `<YOUR-TARGET>` with the desired target or image name. If you are building for `vexpress-qemu`, set the target to `core-image-full-cmdline`. If you are building for the `beaglebone`, set the target to `core-image-base`. For more information about the differences with image types on the BeagleBone Black please see [the official Yocto Project BeagleBone support page](https://www.yoctoproject.org/downloads/bsps/pyro23/beaglebone?target=_blank).
+!!! Please replace `<YOUR-TARGET>` with the desired target or image name. If you are building for `vexpress-qemu`, set the target to `core-image-full-cmdline`. If you are building for the `beaglebone`, set the target to `core-image-base`. For more information about the differences with image types on the BeagleBone Black please see [the official Yocto Project BeagleBone support page](https://www.yoctoproject.org/downloads/bsps/rocko24/beaglebone?target=_blank).
```
Comment (felixge) [inline 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: Is `core-image-base` also the recommended target for RPI3?

Comment (drewmoseley) [reply 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: That's a good question.  I normally use core-image-full-cmdline for all my local builds.  @kacf is there a reason we are specifying core-image-base here?

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md [hunk 2]
```diff
@@ -87,29 +101,32 @@ source oe-init-build-env
...
-bitbake-layers add-layer ../meta-mender/meta-mender-beaglebone
-```
+
+* Raspberry Pi 3 (other revisions might also work): `bitbake-layers add-layer ../meta-mender/meta-mender-raspberrypi`
+* BeagleBone Black: `bitbake-layers add-layer ../meta-mender/meta-mender-beaglebone`
+* Virtual device: `bitbake-layers add-layer ../meta-mender/meta-mender-qemu`
+
+Other devices may have community support,
+either in [meta-mender](https://github.com/mendersoftware/meta-mender?target=_blank) or other repositories.
+If you are building for a different device, please see [Device integration](../../devices)
```
Comment (drewmoseley) [inline 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: Do we want to mention the meta-raspberrypi layer here?  It is required and the above "add-layer" command will fail as it has a built-in dependency.

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md (hunk 1 above)
Comment (kacf) [reply 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: No, it shouldn't be the recommended target for RPi3.

In fact I don't want `core-image-base` to be the recommended image for Beaglebone either, since `core-image-minimal` and `core-image-full-cmdline` are more stripped down. I believe it was some minor detail that led us to `core-image-base` for Beaglebone, and it never got rectified. Fixing it takes some additional testing though, so I suggest leaving it for now unless you want to go down that rabbit hole.

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md [hunk 3]
```diff
@@ -29,24 +25,42 @@ Each one of these steps can be configured further, see the linked sections for m
 
 The other layers in *meta-mender* provide support for specific boards.
 
-!!! For general information about getting started with Yocto Project, it is recommended to read the [Yocto Project Quick Start guide](http://www.yoctoproject.org/docs/2.3/yocto-project-qs/yocto-project-qs.html?target=_blank).
+!!! For general information about getting started with Yocto Project, it is recommended to read the [Yocto Project Quick Start guide](http://www.yoctoproject.org/docs/2.4/yocto-project-qs/yocto-project-qs.html?target=_blank).
```
Comment (kacf) [inline 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: We have not actually moved to rocko yet. I would put this on a different pull request and merge once we have.

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md (hunk 1 above)
Comment (kacf) [inline 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: Same here, should avoid rocko for the time being, as it's untested.

Comment (drewmoseley) [reply 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: @kacf do you have details on the issue? I've been using core-image-full-cmdline on Beaglebone for some time without issue.

**Comment by GregorioDiStefano** [comment]
(haven't had a chance to look at this, but will do tomorrow)

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md (hunk 1 above)
Comment (estenberg) [reply 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: OK, I am removing this note then!

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md (hunk 2 above)
Comment (estenberg) [reply 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: Good point, added!

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md (hunk 3 above)
Comment (estenberg) [reply 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: Makes sense, I just update these once I get across them and there is a newer one available. This one is not that important (how much in the quickstart will change?), but I removed it as well.

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md (hunk 1 above)
Comment (estenberg) [reply 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: Ack!

**Comment by estenberg** [comment]
Thanks for the reviews, I believe I have updated accordingly now, so I am merging.

File: 04.Artifacts/01.Building-Mender-Yocto-image/docs.md (hunk 1 above)
Comment (kacf) [reply 04.Artifacts/01.Building-Mender-Yocto-image/docs.md]: > do you have details on the issue?

No, sorry, I don't remember what the issue was. But I suspect there is no issue anymore, because it was something very specific that I think I recall was fixed at some point. So the obstacle is just to make sure that all references we have in the build scripts are amended to point to the new image. Not hard at all, but like everything else, it's work. :-)
